src/commands/whoami.rs

```diff
@@ -2,11 +2,17 @@ use crate::settings::global_user::GlobalUser;
 use crate::terminal::{emoji, message};
 
 pub fn whoami(user: &GlobalUser) {
-    let msg = format!(
-        "{} You are logged with the email '{}'.",
-        emoji::WAVING,
-        user.email
-    );
+    // Only returns email if present (using email + API key for auth).
+    match &user.email {
+        Some(email) => {
```

Comment:
this should probably be a pair of if lets that handle four cases:

email + api key
api token

email + api key + api token. 

unless we explicitly unset email + api keys when people run `wrangler config --token`.

Reply:
it does explicitly unset.